test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,390 @@
+
+	customDnsServiceName = "custom-dns-service"
+	customDnsConfigName  = "custom-dns-config"
+	customDnsPort        = 53
+	customDnsTargetPort  = 53
+
+	customDnsImage         = "coredns/coredns:latest"
+	customDnsPodName       = "custom-dns-server"
+	customDnsContainerName = "coredns"
+	customDnsLabelKey      = "app"
+	customDnsLabelValue    = "custom-dns"
+	customDnsVolume        = "config-volume"
+
+	fqdnPolicyName            = "test-acnp-fqdn"
+	fqdnPodSelectorLabelKey   = "app"
+	fqdnPodSelectorLabelValue = "fqdn-cache-test"
+	toolBoxPodName            = "toolbox"
+
+	agnHostPort          = 80
+	agnHostLabelKey      = "app"
+	agnHostLabelValue    = "agnhost"
+	agnHostPodNamePreFix = "agnhost-"
+)
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
+	skipIfAntreaPolicyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+
+	customDnsService, err := createCustomDnsService(data)
+	require.NoError(t, err, "Error when creating custom DNS service: %v", err)
+
+	dnsServiceIP := getCustomDnsServiceIP(t, data, customDnsService)
+
+	setDnsServerAddressInAntrea(t, data, dnsServiceIP)
+
+	// I think creating 2 pods with a deployment might be better.
+	createHttpAgnhostPod(t, data, randName(agnHostPodNamePreFix), map[string]string{agnHostLabelKey: agnHostLabelValue})
+	createHttpAgnhostPod(t, data, randName(agnHostPodNamePreFix), map[string]string{agnHostLabelKey: agnHostLabelValue})
+
+	k8sUtils, err = NewKubernetesUtils(data)
+	require.NoError(t, err, "error getting k8s utils %+v", err)
+
+	agnHostPods, err := k8sUtils.GetPodsByLabel(data.testNamespace, agnHostLabelKey, agnHostLabelValue)
+	require.NoError(t, err, "error getting Pods by label  %+v", err)
+
+	//domainMapping holds whether the IP is mapped to Domain or not.
+	domainMapping := make(map[string]bool)
+
+	// pick an IP to be added in config
+	var ipForDnsConfig string
+	for idx, pod := range agnHostPods {
+		ipStr := strings.TrimSpace(pod.Status.PodIP)
+		domainMapping[ipStr] = false
+		//pick last IP for config
+		if idx == len(agnHostPods)-1 {
+			ipForDnsConfig = ipStr
+		}
+	}
+
+	dnsConfigData := createDnsConfig(ipForDnsConfig)
+
+	customDnsConfigMapObject, err := data.CreateConfigMap(data.testNamespace, customDnsConfigName, dnsConfigData, nil, false)
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+	domainMapping[ipForDnsConfig] = true
+
+	createDnsPod(t, data)
+
+	builder := buildFqdnPolicy(t)
+
+	defer tearDown(t, data, builder)
+
+	createToolBoxPod(t, data, dnsServiceIP)
+
+	// idea copied from antctl_test.go:284
+	// getEndpointStatus will return "Success", "Failure", or the empty string when out is not a
+	// marshalled metav1.Status object.
+	getEndpointStatus := func(out []byte) string {
+		var status metav1.Status
+		if err := json.Unmarshal(out, &status); err != nil {
+			// Output is not JSON or does not encode a metav1.Status object.
+			return ""
+		}
+		return status.Status
+	}
```

Comment:
I don't think this makes sense here. Are you sure you are accessing an endpoint that returns a serialized `metav1.Status` value?

If you are using `agnhost netexec` for your server Pod, it returns a timestamp: https://pkg.go.dev/k8s.io/kubernetes/test/images/agnhost/netexec

IMO, you don't care about the contents of the HTTP response here, you only care about the exit code of the curl command (`err` value returned by `RunCommandFromPod`)